pkg/operator/controllers/controller_utils.go

```diff
@@ -71,6 +72,18 @@ func randString(n int) string {
 	return string(b)
 }
 
+// Determine whether to use an existing saved time variables or the current time for state duration
+func getStateDuration(endTime *metav1.Time, startTime *metav1.Time) float64 {
+	var stateDuration float64
+	// If the end time state is missing, use the current time
+	if endTime == nil {
```

Comment:
Could we make this

```golang
if endTime == nil {
		return math.Abs(float64(time.Since(startTime.Time).Milliseconds()))
	} 
 return math.Abs(float64(endTime.Time.Sub(startTime.Time).Milliseconds()))
```

to minimize indenting?